### PR TITLE
Issue #138: Fix user signup in test setup

### LINKED ISSUE #138: As a dev, I want to create and clean up a user for each test
## Description

**Context**  
Currently, in our test suite, we use a pre-created user to run the tests. This approach can lead to unintended side effects and dependencies across tests. Additionally, creating a user also creates a blob container, which needs to be properly cleaned up after each test to avoid resource leakage.

**Problem Statement**  
The current use of a pre-created user in tests can cause conflicts and unreliable test results. Furthermore, the lack of a cleanup mechanism for the blob container results in cluttered and unmanaged resources in the test environment.

**Acceptance Criteria**
- [ ] Each test should dynamically create a new user using a setup function.
- [ ] A new blob container should be created whenever a user is created.
- [ ] A cleanup function, such as `delete_user`, should be implemented to remove the user and associated blob container after each test.
- [ ] All tests should be refactored to use the new setup and cleanup functions, ensuring no reliance on pre-created users.

### PR DESCRIPTION
**Pull request by snakedye**
A test user is signed up in the setup function.

Closes #138

### CODE REVIEW

File: tests/test_app.py [hunk 1]
```diff
@@ -39,6 +39,14 @@ def credentials(cls, username, password) -> str:
     def setUp(self):
         app.testing = True
         self.client = app.test_client()
+        # self.client.post(
```
Comment (k-allagbe) [inline tests/test_app.py:42]: Incomplete? The requirements are not satisfied.

**Comment by snakedye** [comment]
One pending issue is that the container for each users are not deleted when the transactions are rolled back.

File: tests/test_app.py [hunk 2]
```diff
@@ -53,14 +53,15 @@ def tearDown(self):
         connection_manager.rollback()
         connection_manager.put()
 
-    
+    @classmethod
     def tearDownClass(self):
         # Delete the content of the storage account
         FERTISCAN_STORAGE_URL = os.getenv("FERTISCAN_STORAGE_URL")
         with ContainerClient.from_connection_string(
             FERTISCAN_STORAGE_URL, container_name=f"user-{self.username}"
```
Comment (k-allagbe) [inline tests/test_app.py:61]: It takes the user id instead of the username. You can save the user id from the `/signup` response in `setUp`.

**Review by k-allagbe — APPROVED** [review]
Tested, working as expected. Please consider writing a quick script to clean up the storage.

**Comment by snakedye** [comment]
@k-allagbe if we just delete all the containers we might delete some of existing users. We'd need to do both at the same time.

**Comment by k-allagbe** [comment]
> @k-allagbe if we just delete all the containers we might delete some of existing users. We'd need to do both at the same time.

I was not suggesting it as part of this issue. Just some side chore if you will.

**Pull request merged**